### PR TITLE
Enhancement/graphviz other formats

### PR DESCRIPTION
**Pull request by chuehsien**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [1] adds new tests (if appropriate)
- [1] updates `CHANGELOG.md` (if appropriate)
- [1] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
This PR allows flow.visualize to accept an argument `format` to allow visualization to be produced in formats other than PDF.

## Why is this PR important?
Some downstream apps that take the results of flow.visualize may not be able to interpret PDF easily.

### CODE REVIEW

**Comment by joshmeek** [comment]
Thanks for the PR @chuehsien! Looks like you have one failing test:

```
=================================== FAILURES ===================================
_____ TestFlowVisualize.test_viz_saves_graph_object_with_correct_extension _____

self = <tests.core.test_flow.TestFlowVisualize object at 0x7f5fc2efe090>

    def test_viz_saves_graph_object_with_correct_extension(self):
        import graphviz
    
        tested_formats = ["jpg", "png", "svg", "gif", "jpeg", "pdf", "tif", "tiff"]
        f = Flow(name="test")
        f.add_task(Task(name="a_nice_task"))
    
        with tempfile.TemporaryDirectory() as tmpdir:
            with open(os.path.join(tmpdir, "viz"), "wb") as tmp:
                print(sorted(list(graphviz.FORMATS)))
                for _format in tested_formats:
>                   graph = f.visualize(filename=tmp.name, format=_format)

tests/core/test_flow.py:1242: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.7/site-packages/prefect/core/flow.py:1183: in visualize
    graph.render(filename, view=False, format=format)
/usr/local/lib/python3.7/site-packages/graphviz/files.py:209: in render
    quiet=quiet)
/usr/local/lib/python3.7/site-packages/graphviz/backend.py:221: in render
    run(cmd, capture_output=True, cwd=cwd, check=True, quiet=quiet)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cmd = ['dot', '-Ttif', '-O', 'viz'], input = None, capture_output = True
check = True, encoding = None, quiet = False
kwargs = {'cwd': '/tmp/tmpvwk1avme', 'stderr': -1, 'stdout': -1}
proc = <subprocess.Popen object at 0x7f5fe9407dd0>, out = b''
err = b'Format: "tif" not recognized. Use one of: canon cmap cmapx cmapx_np dot dot_json eps fig gd gd2 gif gv imap imap_np ...on0 mp pdf pic plain plain-ext png pov ps ps2 svg svgz tk vml vmlz vrml wbmp x11 xdot xdot1.2 xdot1.4 xdot_json xlib\n'

    def run(cmd, input=None, capture_output=False, check=False, encoding=None,
            quiet=False, **kwargs):
        """Run the command described by cmd and return its (stdout, stderr) tuple."""
        log.debug('run %r', cmd)
    
        if input is not None:
            kwargs['stdin'] = subprocess.PIPE
            if encoding is not None:
                input = input.encode(encoding)
    
        if capture_output:
            kwargs['stdout'] = kwargs['stderr'] = subprocess.PIPE
    
        try:
            proc = subprocess.Popen(cmd, startupinfo=get_startupinfo(), **kwargs)
        except OSError as e:
            if e.errno == errno.ENOENT:
                raise ExecutableNotFound(cmd)
            else:
                raise
    
        out, err = proc.communicate(input)
    
        if not quiet and err:
            _compat.stderr_write_bytes(err, flush=True)
    
        if encoding is not None:
            if out is not None:
                out = out.decode(encoding)
            if err is not None:
                err = err.decode(encoding)
    
        if check and proc.returncode:
            raise CalledProcessError(proc.returncode, cmd,
>                                    output=out, stderr=err)
E           graphviz.backend.CalledProcessError: Command '['dot', '-Ttif', '-O', 'viz']' returned non-zero exit status 1. [stderr: b'Format: "tif" not recognized. Use one of: canon cmap cmapx cmapx_np dot dot_json eps fig gd gd2 gif gv imap imap_np ismap jpe jpeg jpg json json0 mp pdf pic plain plain-ext png pov ps ps2 svg svgz tk vml vmlz vrml wbmp x11 xdot xdot1.2 xdot1.4 xdot_json xlib\n']
```

**Comment by chuehsien** [comment]
Hi Josh! It passes on my windows machine, but fails on the circleci build? Should exclude tif as a tested format?

**Comment by joshmeek** [comment]
@chuehsien Yeah let's take it out of the tests for now

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ These changes are available in the [master branch](https://github.com/PrefectHQ/
 
 ### Enhancements
 
+- flow.visualize is now able to accept a `format` argument to specify the output file type. [#2447]
```
Comment (joshmeek) [inline CHANGELOG.md:13]:
```suggestion
- `flow.visualize` is now able to accept a `format` argument to specify the output file type - [#2447](https://github.com/PrefectHQ/prefect/issues/2447)
```

**Comment by chuehsien** [comment]
How can i see what went wrong? this is the output on my local machine of the pytest command.

======== 3324 passed, 1405 skipped, 19 xfailed, 33 xpassed, 51 warnings in 182.50s (0:03:02) =========

**Comment by joshmeek** [comment]
> How can i see what went wrong? this is the output on my local machine of the pytest command.
> 
> ======== 3324 passed, 1405 skipped, 19 xfailed, 33 xpassed, 51 warnings in 182.50s (0:03:02) =========

It looks like nothing went wrong in that output! You would see `N failed` if some did

**Pull request merged**